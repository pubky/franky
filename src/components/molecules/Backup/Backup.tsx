'use client';

import { useRouter } from 'next/navigation';

import * as Atoms from '@/atoms';
import * as Molecules from '@/molecules';
import * as App from '@/app';

export const BackupNavigation = () => {
  const router = useRouter();

  const onHandleContinueButton = () => {
    router.push(App.ONBOARDING_ROUTES.HOMESERVER);
  };

  const onHandleBackButton = () => {
    router.push(App.ONBOARDING_ROUTES.PUBKY);
  };

  return (
    <Molecules.ButtonsNavigation
      className="py-6"
      onHandleBackButton={onHandleBackButton}
      onHandleContinueButton={onHandleContinueButton}
      backText="Back"
      continueText="Continue"
    />
  );
};

export const BackupPageHeader = () => {
  return (
    <Atoms.PageHeader>
      <Molecules.PageTitle size="large">
        Back up your <span className="text-brand">pubky.</span>
      </Molecules.PageTitle>
      <Atoms.PageSubtitle>You need a backup to restore access to your account later.</Atoms.PageSubtitle>
    </Atoms.PageHeader>
  );
};
