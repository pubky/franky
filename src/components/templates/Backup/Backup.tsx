import * as Molecules from '@/molecules';

export function Backup() {
  return (
    <Molecules.PageContainer>
      <Molecules.BackupPageHeader />
      <Molecules.BackupMethodCard />
      <Molecules.BackupNavigation />
    </Molecules.PageContainer>
  );
}
