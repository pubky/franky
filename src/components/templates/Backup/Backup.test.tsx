import React from 'react';
import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import { Backup } from './Backup';

// Mock atoms
vi.mock('@/atoms', () => ({
  Container: ({ children, size, className }: { children: React.ReactNode; size?: string; className?: string }) => (
    <div data-testid="container" data-size={size} className={className}>
      {children}
    </div>
  ),
}));

// Mock molecules
vi.mock('@/molecules', () => ({
  BackupPageHeader: () => <div data-testid="backup-page-header">Backup Page Header</div>,
  BackupNavigation: () => <div data-testid="backup-navigation">Backup Navigation</div>,
}));

// Mock organisms
vi.mock('@/organisms', () => ({
  BackupMethodCard: () => <div data-testid="backup-method-card">Backup Method Card</div>,
}));

describe('Backup Template', () => {
  it('renders all main components', () => {
    render(<Backup />);

    expect(screen.getByTestId('backup-page-header')).toBeInTheDocument();
    expect(screen.getByTestId('backup-method-card')).toBeInTheDocument();
    expect(screen.getByTestId('backup-navigation')).toBeInTheDocument();
  });

  it('renders container with correct props', () => {
    render(<Backup />);

    const container = screen.getByTestId('container');
    expect(container).toHaveAttribute('data-size', 'container');
    expect(container).toHaveClass('min-h-dvh');
    expect(container).toHaveClass('gap-6');
    expect(container).toHaveClass('pb-0');
    expect(container).toHaveClass('lg:pb-6');
    expect(container).toHaveClass('pt-4');
    expect(container).toHaveClass('lg:min-h-0');
  });

  it('maintains correct component hierarchy', () => {
    render(<Backup />);

    const container = screen.getByTestId('container');
    const header = screen.getByTestId('backup-page-header');
    const methodCard = screen.getByTestId('backup-method-card');
    const navigation = screen.getByTestId('backup-navigation');

    // All components should be within the container structure
    expect(container).toContainElement(header);
    expect(container).toContainElement(methodCard);
    expect(container).toContainElement(navigation);
  });

  it('renders components in correct order', () => {
    render(<Backup />);

    const container = screen.getByTestId('container');
    const children = Array.from(container.children);

    expect(children).toHaveLength(2);
    const [contentWrapper, navigationWrapper] = children as HTMLElement[];

    expect(contentWrapper).toHaveAttribute('data-testid', 'backup-content');
    const contentChildren = Array.from(contentWrapper.children);
    expect(contentChildren[0]).toHaveAttribute('data-testid', 'backup-page-header');
    expect(contentChildren[1]).toHaveAttribute('data-testid', 'backup-method-card');
    expect(contentWrapper).toHaveClass('flex-1');
    expect(contentWrapper).toHaveClass('lg:flex-none');
    expect(navigationWrapper).toHaveClass('mt-auto');
    expect(navigationWrapper).toHaveClass('onboarding-nav');
    expect(navigationWrapper).toHaveClass('lg:mt-0');
    expect(navigationWrapper.firstChild).toHaveAttribute('data-testid', 'backup-navigation');
  });

  it('follows atomic design structure correctly', () => {
    render(<Backup />);

    // Template should use:
    // - Atoms (Container)
    // - Molecules (BackupPageHeader, BackupNavigation)
    // - Organisms (BackupMethodCard)
    expect(screen.getByTestId('container')).toBeInTheDocument(); // Atom
    expect(screen.getByTestId('backup-page-header')).toBeInTheDocument(); // Molecule
    expect(screen.getByTestId('backup-navigation')).toBeInTheDocument(); // Molecule
    expect(screen.getByTestId('backup-method-card')).toBeInTheDocument(); // Organism
  });

  it('renders without any errors', () => {
    expect(() => render(<Backup />)).not.toThrow();
  });

  it('has proper semantic structure', () => {
    render(<Backup />);

    // Should have a main container
    const container = screen.getByTestId('container');
    expect(container).toBeInTheDocument();

    // Should expose the onboarding content wrapper and navigation wrapper
    expect(container.children).toHaveLength(2);
    expect(container.firstElementChild).toHaveAttribute('data-testid', 'backup-content');
  });
});
