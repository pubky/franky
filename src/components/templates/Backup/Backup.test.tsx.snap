// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Backup Template - Snapshots > matches snapshot 1`] = `
<div
  class="mx-auto w-full flex-col flex container min-h-dvh items-stretch gap-0 px-6 pb-0 pt-4 lg:min-h-0 lg:items-start lg:pb-6"
  data-testid="container"
>
  <div
    class="flex w-full flex-1 flex-col gap-0 lg:flex-none"
    data-testid="backup-content"
  >
    <div
      data-testid="backup-page-header"
    >
      Backup Page Header
    </div>
    <div
      data-testid="backup-method-card"
    >
      Backup Method Card
    </div>
  </div>
  <div
    class="onboarding-nav mt-auto w-full lg:mt-0"
  >
    <div
      data-testid="backup-navigation"
    >
      Backup Navigation
    </div>
  </div>
</div>
`;
